pkg/interceptors/recovery.go

```diff
@@ -0,0 +1,47 @@
+package interceptors
+
+import (
+	"context"
+	stdlog "log"
+	"runtime/debug"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+	grpcrecovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	sdkloggercontext "github.com/scribd/go-sdk/pkg/context/logger"
+)
+
+// RecoveryUnaryServerInterceptor returns a unary server interceptor that recovers from panics,
+// sends a sentry event, log in fatal level and halts the service.
+// IMPORTANT: This interceptor should be the last one in the interceptor chain.
+func RecoveryUnaryServerInterceptor() grpc.UnaryServerInterceptor {
+	return grpcrecovery.UnaryServerInterceptor(recoveryOption...)
+}
+
+// RecoveryStreamServerInterceptor returns a streaming server interceptor that recovers from panics,
+// sends a sentry event, log in fatal level and halts the service.
+// IMPORTANT: This interceptor should be the last one in the interceptor chain.
+func RecoveryStreamServerInterceptor() grpc.StreamServerInterceptor {
+	return grpcrecovery.StreamServerInterceptor(recoveryOption...)
+}
+
+var recoveryOption = []grpcrecovery.Option{
+	grpcrecovery.WithRecoveryHandlerContext(func(ctx context.Context, rec interface{}) (err error) {
+		sentry.CurrentHub().Recover(rec)
+		sentry.Flush(time.Second * 5)
```

Comment:
Shouldn't this be wrapped in:

```
if rec := recover(); rec != nil {
}
```

I'm slightly confused as to how this is supposed to work for gRPC. 🤔

Reply:
It is implemented in the [go-grpc-middleware](https://github.com/grpc-ecosystem/go-grpc-middleware/blob/62b7de50cda5a5d633f1013bfbe50e0f38db34ef/interceptors/recovery/interceptors.go#L25-L36) library when passing the option with `grpcrecovery.Option`.

Reply:
Got it – thanks for explaining.